app/components/collections/Thumbnails.tsx

```diff
@@ -1,38 +1,78 @@
-import { faMap, faTableCells } from "@fortawesome/free-solid-svg-icons";
+import { faMap, faTableCells, faList } from "@fortawesome/free-solid-svg-icons";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
-import { Link } from "@remix-run/react";
-import type { ReactNode } from "react";
+import { Link, useSearchParams, useNavigate } from "@remix-run/react";
```

Comment:
This is where your error is coming from. You can only have `useNavigate` in a function that is a route. It doesn't look like you're using it, so you should just be able to delete it here.